### PR TITLE
Chef 13 compatibility fixes

### PR DESCRIPTION
**Pull request by jcorme**
Some notes:

* One unit test fails on Chef 12 because of [this](https://github.com/osuosl-cookbooks/osl-docker/pull/22/files#diff-9f91e0135de8f3ef2e8493be0f54648bR242) change. On Chef 13, the key property of apt_repository  can be a String or an Array, but it is [stored as an array](https://github.com/chef/chef/pull/6372/files#diff-a8088b57696c6dfb39ee6d2e03f33d76R36), which means that the property in the test has to be an array. With Chef 12, the key property can not be an Array and is stored as a String...not sure what the best course of action is.

* I replaced the yum-docker cookbook with chef-yum-docker. yum-docker is abandoned (at least AFAICT...the GitHub repo has been deleted) and uses an old version of the yum cookbook that causes deprecation warnings. Hopefully this also fixes some of the dependency conflict issues I've been having with the osl-acme cookbook, which has been a big pain.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -16,25 +16,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 if node['platform_family'] == 'rhel' && node['kernel']['machine'] != 's390x'
-  include_recipe 'yum-docker'
+  include_recipe 'chef-yum-docker'
```
Comment (ramereth) [inline recipes/default.rb]: Which this change, we need to also remove the old ``docker-main`` repository.

File: recipes/default.rb [hunk 2]
```diff
@@ -43,7 +38,7 @@
   only_if { node['platform_family'] == 'debian' && node['platform_version'].to_i >= 9 }
 end
 
-apt_repository 'docker-main' do
+apt_repository 'docker-stable' do
```
Comment (ramereth) [inline recipes/default.rb]: This should probably be converted to using the ``chef-apt-docker`` cookbook instead.

File: recipes/workstation.rb [hunk 3]
```diff
@@ -17,8 +17,4 @@
 # limitations under the License.
 node.override['osl-docker']['host'] = 'tcp://127.0.0.1:2375'
 
-magic_shell_environment 'DOCKER_HOST' do
-  value 'tcp://127.0.0.1:2375'
-end
-
```
[the remark this answers is not in the document]
Comment (jcorme) [reply recipes/workstation.rb]: This duplicates the [resource defined in the default recipe](https://github.com/osuosl-cookbooks/osl-docker/pull/22/files/d7cb42f3ed0296b88e734bb5229366d624fd482b#diff-55bf87238c9b8af164c5133a60721f12L118).

**Comment by jcorme** [comment]
!bump major *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a major-level version bump to v2.0.0.  Have a nice day!

**Pull request merged**